src/test/java/io/debezium/connector/spanner/PgDataTypesIT.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @AfterAll
+    static void clear() throws InterruptedException {
+        pgDatabaseConnection.dropChangeStream(pgsqlChangeStreamName);
+        pgDatabaseConnection.dropTable(pgsqlTableName);
+    }
+
+    @Test
+    public void shouldStreamUpdatesToKafkaWithTheCorrectType()
+            throws InterruptedException, ExecutionException {
+        final Configuration config = Configuration.copy(basePgConfig)
+                .with("gcp.spanner.change.stream", pgsqlChangeStreamName)
+                .with("name", pgsqlTableName + "_test")
+                .with("gcp.spanner.start.time",
+                        DateTimeFormatter.ISO_INSTANT.format(Instant.now()))
+                .build();
+
+        initializeConnectorTestFramework();
+        start(SpannerConnector.class, config);
+        assertConnectorIsRunning();
+        final long insertedRows = pgDatabaseConnection.executeUpdate("INSERT INTO " + pgsqlTableName
+                + "(id"
+                + ", boolcol"
+                + ", int64col"
+                + ", float64col"
+                + ", timestampcol"
+                + ", datecol"
+                + ", stringcol"
+                + ", bytescol"
+                + ", arrcol"
+                // + ", float32col"
+                // + ", numericcol"
+                // + ", jsoncol"
+                + ") "
+                + "VALUES (1"
+                + ", true"
+                + ", 42"
+                + ", 2.71"
+                + ", '1970-01-01 00:00:00 UTC'"
+                + ", '1970-01-01'"
+                + ", 'stringVal'"
+                + ", bytea 'bytesVal'"
+                // + ", 3.14"
+                // + ", 6.023"
+                // + ", JSONB '\"Hello\"'"
+                + ", ARRAY['a', 'b']"
+                + ")");
+
+        assertEquals(1, insertedRows);
+
+        assertTrue(waitForAvailableRecords(waitTimeForRecords(), TimeUnit.SECONDS));
+        SourceRecords sourceRecords = consumeRecordsByTopic(10, false);
+        List<SourceRecord> records = sourceRecords.recordsForTopic(getTopicName(config, pgsqlTableName));
+        assertThat(records).hasSize(1);
+
+        Struct record = (Struct) (records.get(0).value());
+        assertThat(record.get("op")).isEqualTo("c");
+        assertThat(record.schema().field("after")).isNotNull();
+
+        Struct values = record.getStruct("after");
+
+        assertTrue(values.getBoolean("boolcol"));
+        assertThat(values.getInt64("int64col")).isEqualTo(42);
+        assertThat(values.getFloat64("float64col")).isEqualTo(2.71);
+        assertThat(values.getString("timestampcol")).isEqualTo("1970-01-01T00:00:00Z");
+        assertThat(values.getString("datecol")).isEqualTo("1970-01-01");
+        assertThat(values.getString("stringcol")).isEqualTo("stringVal");
+        assertThat(values.getBytes("bytescol")).isEqualTo("bytesVal".getBytes());
+        // TODO: uncomment after these types are supported.
```

Comment:
Could you please add a comment *who* does not support those types to make it clear?

Reply:
Ack. I've split the TODO into two as there are two reasons why the test is failing. float4 is not yet supported in the emulator, and that's a known issue. 

I do not know why numeric and jsonb are failing, I'm depending on Shuran to help here.